### ISSUE TITLE
Feature Request: Repairing currently feels clunky. Possible future rework?

### ISSUE THREAD

**Issue by Birgliss** [mickelus/tetra#49]
## Feature Request

**Feature description**

Repairing, currently, is kind of repetitive. While I praise Tetra's modularity and boldness, I'm not satisfied with how repairing is currently handled.
I think we could do better. I'm not saying scrap the current system entirely, but we could make improvements.
I don't have anything too specific in mind, but I'm happy to let this be a "brainstorming" thread. So, I'll start a list of complaints I have offhand, with some possible ideas to help alleviate:

- The amount repaired currently feels arbitrary and random.
- Repairing itself is a very slow, GUI-Heavy process. This compounds with the prior point and makes repairing feel a lot worse than it could.

Now, I think we could change a couple things around to make the system more streamlined.
Personally, materials should repair more globally. Nobody likes using 5+ diamonds to repair a pickaxe; however, statically increasing this value could prove uninteresting.
Something that I think would tackle both points on the above list is perhaps a repair "minigame."
Nothing too complex. Something where you give it a material, and you have to perform some form of simple action to repair. The amount repaired will vary based on how well the player does. Perhaps the player could also get a bonus for enchantments on the tool used to repair.

Another thing to consider is giving each "part" of the tool its own durability. However, this could prove too complicated, and cloud up what I think is honestly a pretty charming simplicity.

**How it improves the player experience**

I think this would improve the general flow of the game. In the earlygame, I tend to use stone tools for as long as I can get away with it. You have to repair those frequently, and with the current system, it can take up to 20+ mouse clicks in the same GUI just to get to full durability.
Not only that, making repairing an "enjoyable" experience will help prevent players from being annoyed when their tools break, and possibly even look forward to it, just to repair them. Odd, right?

**Tetra synergies**

Honestly, I don't have a lot for this section. Making the minigames based on each tool you use to repair each given part would make sense, right? Right. That's all I've got.

**Comment by mickelus** [comment]
I agree with you! The initial plan was for all crafting in tetra to be sort of minigames done ingame with the assistance of in-world UI hints, but that proved difficult and i settled with a regular UI _for now_ ™.

I've been playing around with something similar to how "coordinate crafting" works in wildstar https://youtu.be/RVEFwFP_-hk?t=565.

E.g. 
* player provides some material
* player clicks a UI representation of the module that is to be repaired with tools from their inventory
* each click fills up a repair progress bar
* there's a limited amount of clicks before they need to provide more materials
* where the player clicks affects how much progress is gained and how much bonus durability is provided
* tool type, tier & effects could affect progress and durability bonus
* different modules and variants would have different patterns for how much progress & durability bonus is gained, making some variants more difficult to repair and the player would learn and become more efficient over time
* there's preferrably something in common between all variants of a module: if I'm good at repairing  iron hilts I would have a good starting point for repairing copper hilts
* there's preferrably something in common between one variant of different modules: if I'm good at repairing iron hilts I would have a good starting point for repairing iron blades

As you've said, repairing has the potential to be a fun experience and it also has lots of potential to affect the dynamics of the modular item systems. If you have any ideas feel free to pitch in!

The current solution is based on the durability of the modules, but it randomizes the durability gained and the number of materials required for each attempt (between 30-100%). As a first step towards improving this I'll try fixing the amount to 100% of the modules durability and material cost for all repair attempts.

**Comment by Edulml** [comment]
Having each part have its own durability can be a good system, especially if repairing is its own minigame. In the inventory menu we can already press shift to get information about all modules of the tool so durability could also be added, having its text in a different color to make it easier to see at a glance.

To me it doesn't make much sense to spend materials to fix stuff such as counterweights or socketed gems, it almost feels like you are replacing them completely. Part of the minigame could be reseating such parts properly, or resharpening the blade/tool. And if a part is very neglected it can lose efficiency and it might even fall off the tool, making the entire tool fall apart if it was an essential part such as the handle or a tool head. 

Currently while playing I was very far from my base and to fix my tool I needed to spend lapis lazuli which I didn't have, so I had to remove the socketed gem in order to be able to fix the other ones. As a more immediate solution it would be appreciated if we could fix the parts in the order we wanted to, I won't mind if my tool can't have it's max durability but being unable to repair it when it's sitting at 10 uses left doesn't feel nice.
